suite-native/atoms/src/Button/Button.tsx

```diff
@@ -1,118 +1,196 @@
+const textSizeToVariantMap = {
+    small: 'hint',
+    medium: 'body',
+    large: 'body',
+} as const satisfies Record<ButtonSize, TypographyStyle>;
+
+const iconStyle = prepareNativeStyle((utils, { position }: IconStyleProps) => ({
+    extend: [
+        {
+            condition: position === 'left',
+            style: {
+                marginRight: utils.spacings.small,
             },
-            red: {
-                backgroundColor: utils.transparentize(0.9, utils.colors.red),
+        },
+        {
+            condition: position === 'right',
+            style: {
+                marginLeft: utils.spacings.small,
             },
-        };
+        },
+    ],
+}));
 
+export const buttonStyle = prepareNativeStyle<ButtonStyleProps>(
+    (utils, { size, colorScheme, isDisabled, isPressed }) => {
+        const schemeColors = buttonSchemeToColorsMap[colorScheme];
         return {
             flexDirection: 'row',
             justifyContent: 'center',
             alignItems: 'center',
             borderRadius: utils.borders.radii.round,
-            ...buttonSizeStyles[size],
-            ...buttonColorSchemeStyles[colorScheme],
-            extend: {
-                condition: isDisabled,
-                style: {
-                    backgroundColor: utils.colors.gray200,
+            backgroundColor: utils.colors[schemeColors.backgroundColor],
+            extend: [
+                {
+                    condition: isDisabled,
+                    style: {
+                        backgroundColor: utils.colors[schemeColors.disabledBackgroundColor],
+                    },
                 },
-            },
+                {
+                    condition: isPressed,
+                    style: {
+                        backgroundColor: utils.colors[schemeColors.onPressColor],
+                    },
+                },
+                {
+                    condition: size === 'small',
+                    style: {
+                        height: 40,
+                        paddingVertical: 10,
+                        paddingHorizontal: utils.spacings.medium,
+                    },
+                },
+                {
+                    condition: size === 'medium',
+                    style: {
+                        height: 48,
+                        paddingVertical: 12,
+                        paddingHorizontal: 20,
+                    },
+                },
+                {
+                    condition: size === 'large',
+                    style: {
+                        height: 56,
+                        paddingVertical: utils.spacings.medium,
+                        paddingHorizontal: utils.spacings.large,
+                    },
+                },
+            ],
         };
     },
 );
 
-const buttonColorSchemeFontColor: Record<ButtonColorScheme, Color> = {
-    primary: 'gray0',
-    gray: 'gray800',
-    red: 'red',
-};
-
 export const Button = ({
```

Comment:
Please add onPressIn/Out animations for both buttons.

Reply:
animations created in 0f7794d

Reply:
fixed in 0f7794d